#High-priority Order #1

#Number of orders: 2
#Number of shipments: 1 per order
#Challenges:
#  -Interruption: Yes
#  -Drop parts: No
#  -Sensor blackout: No
#  -Part flipping: No
#  -Faulty parts: Yes
#  -Moving obstacle: No

#Containers
# -Belt: No
# -Bins: Yes
# -Shelf: Yes (shelf1 & shelf2)
#AGV
# -order_0:
#   -shipment_0: agv1
#Time limit: No

time_limit: 500

options:
  gazebo_state_logging: true
  insert_models_over_bins: true
  insert_models_over_shelves: true
  belt_population_cycles: 0
  spawn_extra_models: true
  model_type_aliases:
    order_part1: gasket_part_red
    order_part2: gear_part_green
    order_part3: disk_part_blue
    order_part4: disk_part_red
    order_part5: disk_part_green

shelf_layout:
  row_1: ['base','collar',0,'base'] #shelves: 3, 4, 5
  row_2: ['pipe', 0,'base','base']   #shelves: 6, 7, 8
  row_3: ['base','pipe',0,'collar'] #shelves: 9, 10, 11


orders:
  order_0:
    announcement_condition: time
    announcement_condition_value: 0.0
    shipment_count: 1
    destinations: [agv1]
    products:
      part_0:
        type: gasket_part_red
        pose:
          xyz: [0.0, 0.0, 0]
          rpy: [0, 0, '-pi/4']
      part_1:
        type: gear_part_green
        pose:
          xyz: [-0.15, 0.2, 0]
          rpy: [0, 0, 0]
      part_2:
        type: disk_part_blue
        pose:
          xyz: [0.15, -0.2, 0]
          rpy: [0, 0, '-pi/4']
  order_1:
    announcement_condition: unwanted_products
    announcement_condition_value: 2
    shipment_count: 1
    destinations: [agv2]
    products:
      part_0:
        type: gear_part_green
        pose:
          xyz: [0.0, 0.0, 0]
          rpy: [0, 0, 0]
      part_1:
        type: disk_part_blue
        pose:
          xyz: [0.15, 0.2, 0]
          rpy: [0, 0, 0]
      part_2:
        type: disk_part_red
        pose:
          xyz: [0.15, -0.2, 0]
          rpy: [0, 0, 0]
      part_3:
        type: disk_part_red
        pose:
          xyz: [-0.15, -0.2, 0]
          rpy: [0, 0, 0]
      part_4:
        type: disk_part_green
        pose:
          xyz: [-0.15, 0.2, 0]
          rpy: [0, 0, 0]

models_over_bins:
  bin10:
    models:
      gasket_part_red:
        xyz_start: [0.1, 0.1, 0.0]
        xyz_end: [0.5, 0.5, 0.0]
        rpy: [0, 0, 'pi/4']
        num_models_x: 3
        num_models_y: 3
  bin15:
    models:
      disk_part_blue:
        xyz_start: [0.1, 0.1, 0.0]
        xyz_end: [0.5, 0.5, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 3
        num_models_y: 3
  bin6:
    models:
      disk_part_green:
        xyz_start: [0.1, 0.1, 0.0]
        xyz_end: [0.5, 0.5, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 3
        num_models_y: 3

models_over_shelves:
  shelf1:
    models:
      disk_part_red:
        xyz_start: [1, -0.5, 0.0]
        xyz_end: [0., -0.35, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 5
        num_models_y: 2
  shelf2:
    models:
      gear_part_green:
        xyz_start: [1, 0.5, 0.0]
        xyz_end: [0., 0.4, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 4
        num_models_y: 1

faulty_products:
  - gasket_part_red_1
  - gasket_part_red_5
  - disk_part_blue_1
  - disk_part_blue_2
  - disk_part_blue_3
  - gear_part_green_2
